mozanalysis/bootstrap.py

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+from functools import partial
+
+
+def _resample(iteration, stat_fn, broadcast_data):
+    np.random.seed()
+    n = len(broadcast_data.value)
+    randints = np.random.randint(0, n, n)
+    return stat_fn(broadcast_data.value[randints])
+
+
+def bootstrap(sc, data, stat_fn, num_iterations=2000, confidence_level=0.95):
```

Comment:
Is documentation of the API going to be a separate step? I'd love to see a docstring here explaining the purpose of the function and describing the parameters (in numpy docs style if you don't have another standard for this project).

Reply:
Yes I'd like to add docstrings. I'll take a look at numpy. Thanks for the suggestion.

Reply:
See https://numpydoc.readthedocs.io/en/latest/

To my knowledge, we tend to use numpy-style docstrings. They're pretty widely used, especially in python data world.

Reply:
FWIW, I've stayed close to standard Sphinx/rST docstring syntax for parameters etc in other data specific projects but I could have missed precedence in some cases.

Reply:
@jezdez has much better context than I on precedent for Python at Mozilla, so do follow his advice here. numpydoc may well be overkill here.

Reply:
Thanks all for the feedback.